system/greenhouse-ccloud/templates/thanos-global.yaml

```
@@ -0,0 +1,76 @@
+{{- if and .Values.thanos.enabled .Values.thanos.globalClusters -}}
+kind: Secret
+apiVersion: v1
+metadata:
+  name: thanos-{{ .Chart.Name }}-tls
+  namespace: {{ .Chart.Name }}
+type: Opaque
+data:
+  ca.crt: {{ required ".Values.thanos.tls.caCert missing" .Values.thanos.tls.caCert | b64enc | quote }} 
+  tls.crt: {{ required ".Values.thanos.tls.tlsCert missing" .Values.thanos.tls.tlsCert | b64enc | quote }} 
+  tls.key: {{ required ".Values.thanos.tls.tlsKey missing" .Values.thanos.tls.tlsKey | b64enc | quote }}
+---
+apiVersion: greenhouse.sap/v1alpha1
+kind: PluginPreset
+metadata:
+  name: thanos-global
+  namespace: {{ .Chart.Name }}
+spec:
+  clusterSelector:
+    matchExpressions:
+    - key: greenhouse.sap/pluginpreset
+      operator: In
+      values:
+      - "true"
+    - key: cluster
+      operator: In
+      values:
+        {{ toYaml .Values.thanos.globalClusters | nindent 8 }}
+{{- if .Values.thanos.globalClusters }}
+  clusterOptionOverrides:
+  {{ include "thanosStoreEndpoints" . | nindent 4 }}
```

I still like the lookup, all in one. feels easy to read

```suggestion
  {{- range $cluster := .Values.thanos.globalClusters -}}
    {{- range $index, $plugin := (lookup "greenhouse.sap/v1alpha1" "Plugin" .Chart.Name "").items }}
      {{- if eq (get $plugin.metadata.labels "greenhouse.sap/plugindefinition") "thanos" }}
        {{- range $index, $option := $plugin.spec.optionValues }}
          {{- if eq $option.name "thanos.query.ingress.grpc.hosts" }}
            {{- $host := (get (index (get $option "value") 0) "host") }}
              {{- range $prefix := $cluster.prefix -}}
                {{- if contains $prefix $host -}}
                  {{- print "- " $host ":443" | nindent 6 }}
                {{- end }}
              {{- end }}
            {{- end }}
          {{- end }}
        {{- end }}
      {{- end }}
    {{- end }}
  {{- end }}
```

---

Granted. But let's continue using values.